src/Rings/mpoly-ideals.jl

```diff
@@ -2009,57 +2009,85 @@ function Base.show(io::IO, I::MPolyIdeal)
-\emph{complete flag variety}$ of $\mathbb{F}^{n}$ by taking `dimensions`
+$\emph{complete flag variety}$ of $\mathbb{F}^{n}$ by taking `dimensions`
 $=(1,\dots,n-1)$ and `n`$=n$.  We remark that evaluating for `F = QQ` yields
 the same set of generators as any field of characteristic $0$.
 
+The first parameter can either be $\mathbb{F}$, or a polynomial ring over $\mathbb{F}$, with $\Sum^{k}_{j=1}{n\choose d_{j}$ variables.  
+The parameter `dimensions` needs to be a vector of distinct increasing entries.
+Evaluating this function with the parameter `minimal = true` returns the reduced Gröbner basis for
+the flag Plücker ideal with respect to the degree reverse lexicographical order.
+
 # Examples
 
 Complete flag variety $\text{Fl}(\mathbb{Q}, (1,2,3), 4)$. 
 ```jldoctest
 julia> flag_pluecker_ideal(QQ,[1,2,3],4)
 Ideal generated by
-  x[[1]]*x[[3, 4]] - x[[3]]*x[[1, 4]] + x[[4]]*x[[1, 3]]
-  -x[[1, 4]]*x[[2, 3]] + x[[2, 4]]*x[[1, 3]] - x[[3, 4]]*x[[1, 2]]
-  x[[2]]*x[[3, 4]] - x[[3]]*x[[2, 4]] + x[[4]]*x[[2, 3]]
-  x[[1]]*x[[2, 4]] - x[[2]]*x[[1, 4]] + x[[4]]*x[[1, 2]]
-  x[[1]]*x[[2, 3]] - x[[2]]*x[[1, 3]] + x[[3]]*x[[1, 2]]
-  x[[1]]*x[[2, 3, 4]] - x[[2]]*x[[1, 3, 4]] + x[[3]]*x[[1, 2, 4]] - x[[4]]*x[[1, 2, 3]]
-  -x[[1, 4]]*x[[1, 2, 3]] + x[[1, 3]]*x[[1, 2, 4]] - x[[1, 2]]*x[[1, 3, 4]]
-  -x[[2, 4]]*x[[1, 2, 3]] + x[[2, 3]]*x[[1, 2, 4]] - x[[1, 2]]*x[[2, 3, 4]]
+  x[[2, 4]]*x[[1, 2, 3]] - x[[2, 3]]*x[[1, 2, 4]] + x[[1, 2]]*x[[2, 3, 4]]
+  x[[1, 4]]*x[[1, 2, 3]] - x[[1, 3]]*x[[1, 2, 4]] + x[[1, 2]]*x[[1, 3, 4]]
   -x[[3, 4]]*x[[1, 2, 3]] - x[[1, 3]]*x[[2, 3, 4]] + x[[2, 3]]*x[[1, 3, 4]]
   -x[[1, 4]]*x[[2, 3, 4]] + x[[2, 4]]*x[[1, 3, 4]] - x[[3, 4]]*x[[1, 2, 4]]
+  -x[[1]]*x[[2, 3, 4]] + x[[2]]*x[[1, 3, 4]] - x[[3]]*x[[1, 2, 4]] + x[[4]]*x[[1, 2, 3]]
+  -x[[1, 4]]*x[[2, 3]] + x[[2, 4]]*x[[1, 3]] - x[[3, 4]]*x[[1, 2]]
+  -x[[1]]*x[[2, 3]] + x[[2]]*x[[1, 3]] - x[[3]]*x[[1, 2]]
+  -x[[2]]*x[[3, 4]] + x[[3]]*x[[2, 4]] - x[[4]]*x[[2, 3]]
+  -x[[1]]*x[[3, 4]] + x[[3]]*x[[1, 4]] - x[[4]]*x[[1, 3]]
+  -x[[1]]*x[[2, 4]] + x[[2]]*x[[1, 4]] - x[[4]]*x[[1, 2]]
 
 ```
 Flag variety $\text{Fl}(\mathbb{Q},(1,3),4)$.
 ```jldoctest
 julia> flag_pluecker_ideal(QQ,[1,3],4)
 Ideal generated by
-  x[[1]]*x[[2, 3, 4]] - x[[2]]*x[[1, 3, 4]] + x[[3]]*x[[1, 2, 4]] - x[[4]]*x[[1, 2, 3]]
+  -x[[1]]*x[[2, 3, 4]] + x[[2]]*x[[1, 3, 4]] - x[[3]]*x[[1, 2, 4]] + x[[4]]*x[[1, 2, 3]]
+```
+An example with a custom ring as input.
+```jldoctest
+julia> R, _ = polynomial_ring(QQ, 8)
+(Multivariate polynomial ring in 8 variables over QQ, QQMPolyRingElem[x1, x2, x3, x4, x5, x6, x7, x8])
+
+julia> flag_pluecker_ideal(R, [1,3], 4; minimal=false)
+Ideal generated by
+  -x1*x6 + x2*x5 - x3*x7 + x4*x8
 ```
 """
-function flag_pluecker_ideal(F::Field, dimensions::Vector{Int}, n::Int)
+function flag_pluecker_ideal(F::Field, dimensions::Vector{Int}, n::Int; minimal::Bool=true)
+  L = sum(binomial(n, d) for d in dimensions)
+  ring, _ = polynomial_ring(F, L; cached=false)
```

Comment:
I think you now want to change this back to a reduce-vcat-subsets thing, as you want to use the actual subsets for the variable naming in the poly ring